### PR TITLE
Fix warnings when building with Clang

### PR DESCRIPTION
**Pull request by vanwinkeljan**
This PR contains a number of fixes for warnings when running sanitycheck for native_posix with clang v8.0.

### CODE REVIEW

File: arch/posix/CMakeLists.txt [hunk 1]
```diff
@@ -10,7 +10,6 @@ else ()
 endif ()
 
 zephyr_compile_options(
-  -MMD
```
Comment (aescolar) [inline arch/posix/CMakeLists.txt]: These 2 lines seem to be just vestigial from the pre cmake era.
@SebastianBoe Should we remove both this and the `-MP` below?

**Review by aescolar — COMMENTED** [review]
By the way, I have no other obvious concern with the changes on this PR,
but @jukkar should bless the net changes

File: arch/posix/CMakeLists.txt (hunk 1 above)
Comment (SebastianBoe) [reply arch/posix/CMakeLists.txt]: Probably. I'd diff the build directory with and without these flags to verify that this is safe.

The flags were first added in commit 76f76441188df11a6b536f0725a2220ccdc503ee

**Review by jukkar — APPROVED** [review]
+1 to networking changes

File: arch/posix/CMakeLists.txt (hunk 1 above)
Comment (vanwinkeljan) [reply arch/posix/CMakeLists.txt]: Removed -MP

File: samples/net/sockets/packet/src/packet.c [hunk 2]
```diff
@@ -29,6 +27,8 @@ struct packet_data {
 	struct k_delayed_work send;
 };
 
+static struct packet_data packet;
```
Comment (pfalcon) [inline samples/net/sockets/packet/src/packet.c]: Please elaborate message for this commit. "Moved declaration of packet_data before definition" doesn't explain what happens here. "Move declaration of struct packet_data before definition of a variable of its type" would.

**Review by SebastianBoe — APPROVED** [review]

File: samples/net/sockets/packet/src/packet.c (hunk 2 above)
Comment (vanwinkeljan) [reply samples/net/sockets/packet/src/packet.c]: done

**Review by pfalcon — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Pull request merged**